### PR TITLE
Feature/http api accessing

### PR DESCRIPTION
**Pull request by BirukouAliaksei**
Add client api for set and get data by key and tests

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			_, err := w.Write([]byte(j))
+			require.NoError(t, err)
+
+		}))
+		defer s.Close()
+
+		c := New(s.URL)
+		_, err := c.GetByKey(ctx, getter, method, cipherKey)
+		require.Error(t, err)
+		require.EqualValues(t, "secret client: can't get data: body: \"test bad request body\", status code: 400", err.Error())
```
Comment (alesanro) [inline pkg/client/client_test.go:156]: When checking for unexpected status code your server's hander responses with `j` then the body in returned error must also contain full body in its description (not its part).

**Review by alesanro — CHANGES_REQUESTED** [review]
As a general rule: always check your pull request (mean, review it by your own) because you may find something that should be changed.

I also should notice that it is possible to make Pull requests as `Draft` - that will signal that this PR is not yet ready. And when it is changed from `Draft` to normal PR - it means that a reviewer could review in any moment since then.

File: pkg/client/client_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+func TestClient_SetByKey(t *testing.T) {
+	cipherKey := "c-key"
+	getter := "key"
+	method := "cloud"
+	value := "test-value"
+	t.Run("success", func(t *testing.T) {
+		ctx := context.Background()
+		s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			require.EqualValues(t, "/", r.URL.Path)
```
[the remark this answers is not in the document]
Comment (BirukouAliaksei) [reply pkg/client/client_test.go:24]: added require get/post methods on next line

File: pkg/client/client.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	options options
+	url     string // address where client will be work with server
+}
+
+// options client
+type options struct {
+	c *http.Client
+}
+
+var defaultClient = options{
```
Comment (alesanro) [inline pkg/client/client.go:26]:
```suggestion
var defaultClient = options{
```

Comment (alesanro) [inline pkg/client/client.go:26]: It is not a client but options
```suggestion
var defaultOptions = options{
```

File: pkg/client/client.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	c: &http.Client{Timeout: 20 * time.Second},
+}
+
+type Option func(o *options)
+
+// Client provide any http client
+func Client(c http.Client) Option {
```
Comment (alesanro) [inline pkg/client/client.go:33]: No need to copy `http.Client` if we make use of a pointer.
```suggestion
func Client(c *http.Client) Option {
```

File: pkg/client/client_test.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+			require.EqualValues(t, http.MethodPost, r.Method)
+			require.EqualValues(t, r.Header.Get(api.ParamCipherKey), cipherKey)
```
Comment (alesanro) [inline pkg/client/client_test.go:26]: Assume that expected value should be `cipherKey` and actual `r.Header.Get(api.ParamCipherKey)`

Look thought all tests, find and fix the same issues

```suggestion
			require.EqualValues(t, cipherKey, r.Header.Get(api.ParamCipherKey))
```

**Review by alesanro — CHANGES_REQUESTED** [review]

**Review by alesanro — APPROVED** [review]

**Pull request merged**